packages/lwc/index.d.ts

```diff
@@ -194,39 +194,56 @@ declare module 'lwc' {
      * @param adapter the adapter used to provision data
      * @param config configuration object for the adapter
      */
-    export function wire(
-        adapter: WireAdapterConstructor | LegacyWireAdapterConstructor,
-        config?: WireConfigValue
+    export function wire<Config extends object = any, Value = any, Context extends object = any>(
+        adapter:
+            | WireAdapterConstructor<Config, Value, Context>
+            | LegacyWireAdapterConstructor<Config, Value>,
+        config?: WireConfigValue<Config>
     ): PropertyDecorator;
 
-    type LegacyWireAdapterConstructor = (config?: any) => any;
-    type WireConfigValue = Record<string, any>;
-    type ContextValue = Record<string, any>;
+    type LegacyWireAdapterConstructor<Config = any, Value = any> = (config?: Config) => Value;
+    type WireConfigValue<Config extends object = Record<string, any>> = {
+        [K in keyof Config]: Config[K] | string;
+    };
+    type ContextValue<Context extends object = Record<string, any>> = Partial<
```

Comment:
Same as above, `Record<string, any>` is just a subset of `Config extends object`.
```suggestion
    type ContextValue<Context extends object> = Partial<
```

Reply:
Also for backward compat.